### ISSUE TITLE
WAGTAIL_ALLOW_UNICODE_SLUGS returning hyphens/dashes

### ISSUE THREAD

**Issue by Mayanktaker** [wagtail/wagtail#7255]
### Issue Summary

Fresh Install of Wagtail. Trying to convert my Hindi characters in slug url which is auto generated by Wagtail, to English Characters so when I share the url on social media, it dont mess with %%%%. I want to convert my slugs to ascii when make a post/page. I am using WAGTAIL_ALLOW_UNICODE_SLUGS=False
it should convert the unicode Hindi language url to English(like Django Autoslug do). but its showing only ------ in url instead of converted characters.

`http://127.0.0.1:8000/-------------------/ ` instead of http://127.0.0.1:8000/converted-characters-from-my-language/

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. Edit base.py and put `WAGTAIL_ALLOW_UNICODE_SLUGS=False`

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes / no)
Yes

### Technical details

* Python version: 3.9.5
* Django version: 3.2.4
* Wagtail version: 2.13.1
* Browser version:  91.0.4472.101

**Comment by gasman** [comment]
Hi @Mayanktaker - thanks for the report.

This is the intended behaviour - as per #1443, Wagtail does not perform any kind of transliteration on non-ASCII titles. It seems that the overall preference for transliterating vs keeping the original unicode varies between languages (see also https://github.com/wagtail/wagtail/issues/3311#issuecomment-276115890) - but since people who do want to preserve the original text will presumably keep `WAGTAIL_ALLOW_UNICODE_SLUGS` enabled, I'd be happy to accept a PR to add transliteration when WAGTAIL_ALLOW_UNICODE_SLUGS is false. Given that the slug field is usually populated client-side, this would probably have to be a JS solution - [anyascii](https://github.com/anyascii/anyascii) looks like a good fit (we're using the Python version elsewhere in Wagtail).

**Comment by Mayanktaker** [comment]
Hi @gasman ,
Thanks for opening the PR.
I was in Django and Autoslug works fine for me. Now shifting to wagtail but I want transliteration in url. 
Have a look at this [](https://pypi.org/project/django-autoslug/)
Can I use this with wagtail slug ?

**Comment by gasman** [comment]
No, the slug generation happens client-side, so django-autoslug wouldn't be suitable for this. (Also, django-autoslug relies on the [unidecode](https://pypi.org/project/Unidecode/) library for transliteration, and we can't use that because its licence is incompatible with Wagtail's.)

**Comment by andyi95** [comment]
Well, personally I found out a workaround overriding the save method of the page model:

```
from transliterate import translit
from django.utils.text import slugify
import itertools
import re
...

    def save(self, *args, **kwargs):
        if self.slug and not re.search('[а-яА-Я]', self.slug):
            super().save(*args, **kwargs)
            return
        value = translit(self.slug, 'ru', reversed=True)
        slug_candidate = slug_orig = slugify(value)
        for i in itertools.count(1):
            if not BlogPage.objects.filter(slug=slug_candidate).exists():
                break
            slug_candidate = SafeText('{}-{}'.format(slug_orig, i))
        self.slug = slug_candidate
        super().save(*args, **kwargs)
```
